### PR TITLE
Remove server components

### PR DESCRIPTION
**Pull request by KevinHock**
### Description

Now that https://github.com/Yelp/detect-secrets/pull/7 is done, this PR will remove the server components and move them to the, not yet public, [detect-secrets-server](https://github.com/Yelp/detect-secrets-server) repo.

~~In it's current state, it's just displaying what is going to go in the server repo (config files and tests aside). Easier to discuss/fix my mistakes earlier on.~~

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -1,12 +1,8 @@
 aspy.yaml==0.3.0
-boto3==1.4.7
-cached-property==1.3.0
```
Comment (KevinHock) [inline requirements-dev.txt]: So I wasn't 100% sure on removing `cached-property` and `identify` I couldn't find where they were used though so I figured it was boiler-plate.

File: README.md [hunk 2]
```diff
@@ -73,75 +73,8 @@ Hooks available:
...
-
-See the sample `.pysensu.config.yaml.sample` for an example, but be sure to
-name your file `.pysensu.config.yaml`.
-
-#### 5. Installing Crontabs
-
-```
-echo -e "$(crontab -l)\n\n$(detect-secrets-server --initialize)" | crontab -
-```
+Please see the [detect-secrets-server](https://github.com/Yelp/detect-secrets-server)
```
Comment (KevinHock) [inline README.md]: Thoughts on this?

Comment (domanchi) [reply README.md]: Sure.

The alternative is just to remove it altogether, though this option arguably increases SEO.

File: config.yaml.sample [hunk 3]
```diff
@@ -4,5 +4,4 @@ default:
     Base64HighEntropyString: 4.5
```
Comment (domanchi) [inline config.yaml.sample]: Do we have config files in the core, client code?

Comment (KevinHock) [reply config.yaml.sample]: You're right, I missed this one -- deleted.

**Pull request merged**